pyccl/pk2d.py

```diff
@@ -194,71 +227,79 @@ def eval(self, k, a, cosmo):
                 covered by the arrays you passed when initializing this object,
                 the power spectrum will be extrapolated from the earliest
                 available value using the linear growth factor (for which a
-                cosmology is needed).
+                cosmology is needed). If no Cosmology is passed, attempting
+                to evaluate the power spectrum outside of the scale factor
+                boundaries will raise an exception.
 
         Returns:
             float or array_like: value(s) of the power spectrum.
         """
-        # make sure we have growth factors for extrapolation
-        cosmo.compute_growth()
+        # determine if logarithmic derivative is needed
+        if not derivative:
```

Comment:
haha I was thinking in terms of maths, not in terms of language. We first have no derivative, then first derivative. Do you want me to swap them? I don't mind. I think most calls are made with `not derivative == True`.

Reply:
The way I think of `if` statements is that you actually want to put the special cases first. Branch predictors usually evaluate the `else` option first (since that usually covers the most scenarios), so you can improve performance this way.
Of course, this is irrelevant for python, since it's not compiled, but I still like thinking of the `if`, `elseif` parts of an `if` statement that has an `else` clause as the special cases.

Reply:
I agree, but see, special cases are errors, early return statements etc. Here, that is not the case, we just choose between two different options. This if-statement is more comparable to a match-case statement in C.

Reply:
I'm not gonna fight you on this. I'll just change it when you're not looking ;-)